tests/sentry/utils/test_concurrent.py

```diff
@@ -67,3 +71,61 @@ def test_future_broken_callback():
 
     assert callback.call_count == 1
     assert callback.call_args == mock.call(future_set)
+
+
+def test_threaded_executor():
+    executor = ThreadedExecutor(worker_count=1, maxsize=2)
+
+    def waiter(ready, waiting, result):
+        ready.set()
+        waiting.wait()
+        return result
+
+    initial_ready = Event()
+    initial_waiting = Event()
+    initial_future = executor.submit(
+        lambda: waiter(initial_ready, initial_waiting, 1),
+        block=True,
+    )
+
+    initial_ready.wait()  # wait until the worker has removed this item from the queue
+    assert initial_future.running()
+
+    low_priority_ready = Event()
+    low_priority_waiting = Event()
+    low_priority_future = executor.submit(
+        lambda: waiter(low_priority_ready, low_priority_waiting, 2),
+        block=True,
+        priority=10,
+    )
+
+    high_priority_ready = Event()
+    high_priority_waiting = Event()
+    high_priority_future = executor.submit(
+        lambda: waiter(high_priority_ready, high_priority_waiting, 3),
+        block=True,
+        priority=0,
+    )
+
+    queue_full_future = executor.submit(lambda: None, block=False)
+    with pytest.raises(Full):
+        queue_full_future.result()
```

Comment:
This (and any call to `result` below) could hypothetically block the test suite forever if something weird happens, I should probably set an arbitrarily high timeout here just to be careful.

Reply:
Couldn't hurt, but it seems like it'll fail (er, timeout tests/travis) now or never, really? I guess if it's easy enough.

Reply:
I cleaned this up in 4171b545b7299dae09ea2919813c9907bfdf6461 for safety/sanity's sake.